linera-chain/src/data_types.rs

```diff
@@ -880,7 +930,9 @@ impl ExecutedBlock {
     }
 
     pub fn required_blob_ids(&self) -> HashSet<BlobId> {
-        self.outcome.required_blob_ids()
+        let mut required_blob_ids = self.outcome.required_blob_ids();
+        required_blob_ids.extend(self.block.application_blobs_from_incoming_messages());
```

Comment:
Currently getting the application blobs from incoming messages like this, but we could also add these blobs as oracle responses whenever we have an incoming message to that application 🤔 but not sure